src/middleware/auth.js

```diff
@@ -0,0 +1,35 @@
+import { parse as uuidParse } from 'uuid';
+import db from '../db';
+
+const authMiddleware = (req, res, next) => {
+  const userId = req.get('X-User-Id');
+  if (!userId) {
+    return res.status(401).json({ error: 'Unauthorized ' });
+  }
+
+  try {
+    uuidParse(userId);
+  } catch (e) {
+    return res.status(401).json({ error: 'Invalid userId' });
+  }
```

Comment:
Both cases should actually never happen as the service is called from the API gateway. It's good that you catch those cases, but as there appearance would indicate a **serious** issue with the infrastructure, we should add a *error level log* here.